spinalcordtoolbox/aggregate_slicewise.py

```diff
@@ -232,6 +232,13 @@ def aggregate_per_slice_or_level(metric, mask=None, slices=[], levels=[], persli
     :param map_clusters: list of list of int: See func_map()
     :return: Aggregated metric
     """
+    if vert_level:
+        vert_level_slices = Image(vert_level).change_orientation('RPI').data.shape[2]
+        metric_slices = metric.data.shape[0]
+        if vert_level_slices != metric_slices:
```

Comment:
i'll approve it (we have other priority stuff to deal with), but the test could be made more sensitive. e.g., if a user inputs a vertebral level file from another region/subject/whatever that happens to have the same number of slices as the metric, processing will go on but will be wrong. Alternatively, the qform and XYZ dimensions of the image header could be compared (ie: to make sure both volumes are in the exact same coordinate system).

Reply:
Good points. I'll unlink the issue so that it stays open, then add your comments here to the issue description.